hack/vuln-scan-local.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Karmada Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This script starts a images scanning with trivy
+# Parameters: [skip-image-generation] if you want to skip the image generation step to make scanning faster
```

Comment:
Can you remind me what's the use case that people don't want to build the image? Especially for `latest` image?

I can imagine that people might want to scan a released image by a script like:
```
hack/scan-vuln.sh docker.io/karmada/karmada-controller-manager:v1.8.0
```

Reply:
Sometimes, the image is generated before image scanning. For example, after executing the hack/local-up-karmada.sh script to install karmada, I want to do a image scanning, but generating the image takes a long time. And if add the ability to scan the specific image provided which is remote, image generation also need to be skipped .